### ISSUE TITLE
Attachment doesn't show up on iPhone Mail app 

### ISSUE THREAD

**Issue by jhephs** [straydogstudio/axlsx_rails#28]
I don't know where to post this exactly because I can't pinpoint if this is a problem with iOS or with this gem. Sorry about that.

The problem here is that if I use an xlsx template as mail attachment, I can perfectly see it on a browser or another app. However, if I use the default Mail app of iOS, I can tell that it detects that there's an attachment because of the paperclip icon. If I open the email though, it doesn't show the attachment. Any ideas? It would be a great help. Thanks a lot!

P.S. I'm using Rails 4.2 and I have tried both 0.3 and 0.2.1 versions of axlsx_rails.

**Comment by straydogstudio** [comment]
If you forward the email from the phone can you open it elsewhere? 
Or perhaps you can save it to Dropbox and try opening it on the desktop?
If you save the attachment on the desktop, then attach it to a new email, can the phone open it? 
Can the phone open any other xlsx file?
If you use axlsx in a script to create a file, does it do the same thing?

Any of those answers will help determine if it is axlsx, axlsx_rails, or the email template process. Let me know what you find. 

Noel

**Comment by jerielabayon** [comment]
If the email is forwarded from the Mail app, the attachment still doesn't appear in Mail app.

However, if it's forwarded from the browser, the attachment now appears in Mail app. It can now also be opened.

I'm not sure about the other questions. Does any of these help? Thanks

**Comment by straydogstudio** [comment]
Can you post your email / attachment code? 

I'm curious if the phone app still has the correct file even when it can't open it. So if you can forward it from your phone to another email and open it on your desktop, it says the basic structure is there, but there i something the phone doesn't like. 

Are you able to open other spreadsheets on your phone? I believe you will have to have Numbers or the like to view it.

**Comment by jerielabayon** [comment]
Here's the code. I tried forwarding the email from my phone to another email but the attachment doesn't appear in desktop.

The attachment only appears in my phone if I forward it from desktop then open it my phone. Hope you can help. Thanks!!

**app/mailers/admin/report_mailer.rb**
```
class Admin::ReportMailer < ApplicationMailer
  def sample
    attachments['sample.xlsx'] = {
      :mime_type => Mime::XLSX,
      :content => render_to_string(
        :formats => [:xlsx],
        :template => 'admin/reports/sample'
      )
    }

    mail :to => 'jeriel@company.com', :subject => "sample #{Time.now}"
  end
end
```

**app/views/admin/reports/sample.xlsx.axlsx**
```
wb = xlsx_package.workbook

wb.add_worksheet(name: "Content") do |sheet|
  sheet.add_row ['Content']
end
```

**app/views/mailers/report_mailer/sample.html.haml**
```
%h3 sample #{Time.now}
```

**Comment by straydogstudio** [comment]
@jhephs I've never been able to get back to this. Any luck or did you have to abandon it?

**Comment by straydogstudio** [comment]
@jhephs I've not been able to reproduce this bug. Each attachment I create can be opened in the iPhone mail app. I will close this for now. 

It is possibly related to a layout bug that is fixed in 0.5.0. It is also possibly fixed by calling `self.instance_variable_set(:@_lookup_context, nil)` after you set the attachment. If you try either of these and they are successful please let me know.
